internal/ansible/metrics/metrics.go

```diff
@@ -82,6 +84,132 @@ func RegisterBuildInfo(r prometheus.Registerer) {
 	r.MustRegister(buildInfo)
 }
 
+type UserMetric struct {
+	Name      string               `json:"name" yaml:"name"`
+	Help      string               `json:"description" yaml:"description"`
+	Counter   *UserMetricCounter   `json:"counter,omitempty" yaml:"counter,omitempty"`
+	Gauge     *UserMetricGauge     `json:"gauge,omitempty" yaml:"gauge,omitempty"`
+	Histogram *UserMetricHistogram `json:"histogram,omitempty" yaml:"histogram,omitempty"`
+	Summary   *UserMetricSummary   `json:"summary,omitempty" yaml:"summary,omitempty"`
+}
+
+type UserMetricCounter struct {
+	Inc bool    `json:"increment,omitempty" yaml:"increment,omitempty"`
+	Add float64 `json:"add,omitempty" yaml:"add,omitempty"`
+}
+
+type UserMetricGauge struct {
+	Set              float64 `json:"set,omitempty" yaml:"set,omitempty"`
+	Inc              bool    `json:"increment,omitempty" yaml:"increment,omitempty"`
+	Dec              bool    `json:"decrement,omitempty" yaml:"decrement,omitempty"`
+	Add              float64 `json:"add,omitempty" yaml:"add,omitempty"`
+	Sub              float64 `json:"subtract,omitempty" yaml:"subtract,omitempty"`
+	SetToCurrentTime bool    `json:"set_to_current_time,omitempty" yaml:"set_to_current_time,omitempty"`
+}
+
+type UserMetricHistogram struct {
+	Observe float64 `json:"observe,omitempty" yaml:"observe,omitempty"`
+}
+
+type UserMetricSummary struct {
+	Observe float64 `json:"observe,omitempty" yaml:"observe,omitempty"`
+}
+
+func HandleUserMetric(r prometheus.Registerer, metricSpec UserMetric) error {
+	defer recoverMetricPanic()
+	logf.Log.WithName("metrics").Info("Registering", "metric", metricSpec.Name)
+	_, ok := userMetrics[metricSpec.Name]
+	logf.Log.WithName("metrics").Info("at least this worked 1")
```

Comment:
Are we keeping "at least this worked 1" in the code?

Reply:
nope :)